types/model.d.ts

```diff
@@ -246,4 +246,4 @@ export declare class ModelBuilder extends EventEmitter {
  * Union export type for model instance or plain object representing the model
  * instance
  */
-export type ModelData<T extends ModelBase = ModelBase> = T | AnyObject;
+export type ModelData<T extends ModelBase = ModelBase> = T | Partial<T>;
```

Comment:
Strangely enough, `AnyObject` allows arbitrary values not related to ModelBase at all, for example an array. This is a problem in PersistedModel API, where the `data` argument of methods like `create` or `upsert` suddenly allows callers to provide an array, despite the fact that these methods don't support batch mode.

Reply:
We probably should introduce a `DeepPartial` as follows. It's used in TypeORM too.
```ts
export type DeepPartial<T> = { [P in keyof T]?: DeepPartial<T[P]>; };
```

Reply:
wouldn't `T | Partial<T>` be the same as `Partial<T>`?